use RecyclerView instead of EnhancedListView

SoyaLeaf opened the pull request:
I have tested on my Nokia6 and an Android 6.0 device.
It looks like this:
![47d9e71f-be7c-417c-b8e3-c1bd678454e3](https://user-images.githubusercontent.com/20620071/38321305-cce598c4-3869-11e8-9381-8e62bea11761.gif)

iNPUTmice commented:
Cool. I always wanted to do this. Can you bring the background of the swiped element back? (The green one)

iNPUTmice commented:
Oh and please don’t include your gradle.properties in the commit.

SoyaLeaf commented:
> Oh and please don’t include your gradle.properties in the commit.

I added it to the .gitignore long ago , and I don't know why it was committed again, but now I've removed it using `git rm`, hopefully it won't come out again.

Sorry for the trouble. :)

iNPUTmice commented on src/main/java/eu/siacs/conversations/ui/ConversationsOverviewFragment.java:
```diff
@@ -62,6 +72,101 @@
+						,viewHolder.itemView.getRight(),viewHolder.itemView.getBottom(), paint);
+			}
+			int width = viewHolder.itemView.getWidth();
+			viewHolder.itemView.setAlpha(
```
I don’t think we need the swipe to transparency effect. Actually. i know that the previous implementation did it but that was not a conscious choice but just something the library did. Just keep it solid all the way.

SoyaLeaf replied:
Well, I just want it to be consistent with the past. So do I need to delete it?

iNPUTmice replied:
I appreciate the effort but i think it looks better if it stays solid while swiping. The swiped item in Gmail for example stays solid as well.

SoyaLeaf replied:
I have removed it

iNPUTmice commented on src/main/java/eu/siacs/conversations/ui/adapter/ConversationAdapter.java:
```diff
@@ -249,8 +249,17 @@ private void loadAvatar(Conversation conversation, ImageView imageView) {
 	}
 
-	public static class ViewHolder {
-		private View swipeableItem;
```
If we no longer need the swipable item to create the background color can we remove that from the layout file as well?

iNPUTmice commented on gradle.properties:
```diff
@@ -1,15 +1,7 @@
+#Mon Apr 02 22:36:34 CST 2018
+systemProp.http.proxyHost=127.0.0.1
 org.gradle.jvmargs=-Xmx2048M
+systemProp.http.proxyPort=1080
```
Unfortunately your commit still has changes to the gradle.properties. I think a rebase on current master might fix that. (You need to rebase anyway to fix the conflict in the strings.xml file.)

SoyaLeaf replied:
Thanks for guidance, I will try it. 😃

SamWhited replied:
A rebase attempts to carry all changes forward (including these). I would do a `git checkout origin/master gradle.properties`, which will get the current version of this file, then add file and ammend your commit to include it.

iNPUTmice commented:
Merged in f3913ed3648cb70ab39615b5afd5d2f25b3fc8e2
